src/insexpand.c

```diff
@@ -4184,7 +4184,7 @@ get_complete_info(list_T *what_list, dict_T *retdict)
 		    if (compl_curr_match != NULL
 			    && compl_curr_match->cp_number == match->cp_number)
 			selected_idx = list_idx;
-		    if (match->cp_in_match_array)
+		    if (!has_matches || match->cp_in_match_array)
 			list_idx += 1;
```

Comment:
Excuse me. it's not something you fixed and it doesn't change the behavior, but I'd appreciate it if you could fix it with this PR.

```suggestion
			list_idx++;
```

Reply:
Thanks :+1: